src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,100 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../App';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```

Comment:
```suggestion
  const [newMovie, setNewMovie] = useState({
     title: '',
     description: '',
     ...
  })
```
suggest to make one state for all form fields. use event.target.name for change proper field in newMovie state (each input should have name attribute with same value as newMovie appropriate key).